core/src/main/kotlin/com/malinskiy/marathon/analytics/TrackerFactory.kt

```diff
@@ -0,0 +1,72 @@
+package com.malinskiy.marathon.analytics
+
+import com.google.gson.Gson
+import com.malinskiy.marathon.analytics.external.influx.InfluxDbProvider
+import com.malinskiy.marathon.analytics.external.influx.InfluxDbTracker
+import com.malinskiy.marathon.analytics.internal.pub.Track
+import com.malinskiy.marathon.analytics.internal.sub.DelegatingTrackerInternal
+import com.malinskiy.marathon.analytics.internal.sub.ExecutionReportGenerator
+import com.malinskiy.marathon.analytics.internal.sub.TrackerInternal
+import com.malinskiy.marathon.execution.AnalyticsConfiguration.InfluxDbConfiguration
+import com.malinskiy.marathon.execution.Configuration
+import com.malinskiy.marathon.io.FileManager
+import com.malinskiy.marathon.report.allure.AllureReporter
+import com.malinskiy.marathon.report.device.DeviceInfoJsonReporter
+import com.malinskiy.marathon.report.html.HtmlSummaryReporter
+import com.malinskiy.marathon.report.junit.JUnitReporter
+import com.malinskiy.marathon.report.junit.JUnitWriter
+import com.malinskiy.marathon.report.raw.RawJsonReporter
+import com.malinskiy.marathon.report.stdout.StdoutReporter
+import com.malinskiy.marathon.report.test.TestJsonReporter
+import com.malinskiy.marathon.report.timeline.TimelineReporter
+import com.malinskiy.marathon.report.timeline.TimelineSummaryProvider
+import com.malinskiy.marathon.time.Timer
+import org.koin.core.KoinComponent
+import org.koin.core.inject
+import java.io.File
+
+internal class TrackerFactory(private val configuration: Configuration,
+                              private val fileManager: FileManager,
+                              private val gson: Gson,
+                              private val timer: Timer): KoinComponent {
+
+    val track: Track by inject()
```

Comment:
Please use constructor injection instead of field injection

Reply:
done